recipes/libheif/all/conanfile.py

```diff
@@ -0,0 +1,92 @@
+import os
+from conans import ConanFile, tools, CMake
+
+
+class Libheif(ConanFile):
+    name = "libheif"
+    description = "libheif is an HEIF and AVIF file format decoder and encoder."
+    topics = ("conan", "libheif", "heif", "codec", "video")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/strukturag/libheif"
+    license = ("LGPL-3.0-only", "GPL-3.0-or-later", "MIT")
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
```

Comment:
```suggestion
    generators = "cmake", "pkg_config"
```

Reply:
Seems it is not needed because `libheif` use [libfind_pkg_check_modules](https://gitlab.kitware.com/cmake/community/-/wikis/doc/tutorials/How-To-Find-Libraries#using-libfindmacros) macro that use cmake variables provided by cmake wrapper. Local and CCI builds passed without pkg_config generator.

Reply:
Using the `pkgconf` build_requirement without `pkg_config` generator does not make sense.
Then, `pkgconf` might pick up system libraries or do nothing.

Reply:
Yes, `pkgconf` build requirement is not needed too because pkg_config is not used too (dropped it).

Let me explain. `libheif` find modules are smart and they use a combination of `pkg_check_modules` and `find_library`. Even if pkg_config fails [here](https://github.com/strukturag/libheif/blob/master/cmake/modules/FindLibde265.cmake#L2), the following `find_library` call will succeed. But it caused not by hints `LIBDE265_PKGCONF_LIBRARY_DIRS` (they are empty) but by the `CONAN_LIBDE265_ROOT`. So the finder won't use pkg config at all. Compare with `leptonica` recipe.

So the original find modules could have been used in a first time without patching.
The solution suggested by @SpaceIm is totally valid but simply not necessary here.

To avoid picking up system x265 in the same way even when `with_x265` is False, I have added a switch `CMAKE_DISABLE_FIND_PACKAGE_X265`.